src/components/NewEventPage/NewEventForm.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from "react";
+import { FormGroup, InputGroup, NumericInput } from "@blueprintjs/core";
+import { DateRangeInput } from "@blueprintjs/datetime";
+import styles from "./NewEventForm.module.scss";
+import Button from "../UI/Button/Button";
+
+class NewEventForm extends Component {
+  render() {
+    return (
+      <div className={styles.formGroups}>
+        {this.generateNameField()}
+        {this.generateLocationField()}
+        {this.generateAttendanceField()}
+        {this.generateTimingField()}
+        {this.generateSubmitButton()}
+      </div>
+    );
+  }
+
+  generateNameField() {
+    return (
+      <FormGroup
+        className={styles.formGroup}
+        label="Event Name"
+        labelFor="event-name"
+        labelInfo="(required)"
+      >
+        <InputGroup
+          large={true}
+          leftIcon="sort-alphabetical"
+          placeholder="What is your event's name?"
+        />
+      </FormGroup>
+    );
+  }
+
+  generateLocationField() {
+    return (
+      <FormGroup
+        className={styles.formGroup}
+        label="Location"
+        labelFor="location"
+        labelInfo="(required)"
+        helperText="This can be imprecise, e.g. 'Leeds, UK'"
+      >
+        <InputGroup
+          large={true}
+          leftIcon="geolocation"
+          placeholder="Where is your event being held?"
+        />
+      </FormGroup>
+    );
+  }
+
+  generateAttendanceField() {
+    return (
+      <FormGroup
+        className={styles.formGroup}
+        label="Maximum Attendance"
+        labelFor="attendance"
+        labelInfo="(required)"
+      >
+        <NumericInput
+          large={true}
+          fill={true}
+          leftIcon="people"
+          placeholder="What is your event's maximum capacity?"
+        />
+      </FormGroup>
+    );
+  }
+
+  generateTimingField() {
```

Comment:
Is there anyway to break this, and the 3 functions above it, into more modular code? It seems to have a lot of repetition. 🙂

Reply:
Yep, that's what I'm doing next.